runtime-only Dependencies not added to Pitest classpath

**Describe the bug**
Dependencies that are configured to be "runtime only" are not added to Pitest classpath. E.g. when using mockk 1.11.0 it ships transitive dependencies `mockk-agent-jvm` and `kotlin-reflect` with scope "runtime" (see [Maven Central](https://search.maven.org/artifact/io.mockk/mockk/1.11.0/jar)) but Pitest throws 

```
8:50:45 AM PIT >> INFO : MINION : java.lang.NoClassDefFoundError: kotlin/reflect/full/KClasses
8:50:45 AM PIT >> INFO : MINION :       at io.mockk.impl.JvmMockKGateway.<init>(JvmMockKGateway.kt:185)
8:50:45 AM PIT >> INFO : MINION :       at io.mockk.impl.JvmMockKGateway.<clinit>(JvmMockKGateway.kt:173)
```

Verbose output shows that the `mockk-agent-jvm` and `kotlin-reflect` are not part of `classPathElements` passed to Pitest.

**To Reproduce**
Steps to reproduce the behavior:
1. Standard project setup
2. Add `testImplementation("io.mockk:mockk:1.11.0")` dependency
3. Use mockk in one of your tests
4. Test is "green" when executed as unit test but Pitest fails:
    * Fails with "Exception in thread "main" org.pitest.help.PitHelpError: xxx tests did not pass without mutation when calculating line coverage. Mutation testing requires a green suite."
    * shows something like `java.lang.NoClassDefFoundError: Could not initialize class io.mockk.impl.JvmMockKGateway` in verbose output of Pitest

**Expected behavior**
Pitest successfully finishes without exceptions

**Version used:**
`pl.droidsonroids.pitest` `0.2.9`

**Known Workaround**
Discover the dependency that is missing via Pitest verbose output and add it as a "compile" dependency to your project

**Suggested Solution**
Add all entries of `*UnitTestRuntimeClasspath*` configurations (or something like that) to `additionalClasspath`

@koral--  I created[ a branch](https://github.com/koral--/gradle-pitest-plugin/commit/ea9e535fd712474207ac9336f0f226940914a4c7) that showcases the problem. I found out that the problem only occurs in older versions of Mockk prior to 12.x because Mockk fixed their recommended project setup.

With Mockk 11.0.0 (and lower) you will run into the issue I described here if you follow the suggested setup for those releases (without explicitly declaring a `testImplementation` dependency to the `agent-jvm` library). The reason is a missing class:

```
Caused by: java.lang.ClassNotFoundException: io.mockk.proxy.MockKAgentFactory
```

Root cause is that the transitiv dependency `"io.mockk:mockk-agent-jvm:1.11.0"` seems to be just a "runtimeOnly" dependency of mockk (see [Maven Central](https://mvnrepository.com/artifact/io.mockk/mockk/1.11.0) "Runtime dependencies") and is therefore not included in Pitest's classpath.

--- a/src/main/groovy/pl/droidsonroids/gradle/pitest/PitestPlugin.groovy
+++ b/src/main/groovy/pl/droidsonroids/gradle/pitest/PitestPlugin.groovy
@@ -196,6 +196,10 @@ class PitestPlugin implements Plugin<Project> {
             } else if (ANDROID_GRADLE_PLUGIN_VERSION_NUMBER.major == 4) {
                 from(project.configurations["compile"])
                 from(project.configurations["testCompile"])
+            } else if (ANDROID_GRADLE_PLUGIN_VERSION_NUMBER.major > 4) {
+                from(project.configurations["${variant.name}UnitTestRuntimeClasspath"].copyRecursive { configuration ->
+                    configuration.properties.dependencyProject == null
+                })
             }
             from(project.configurations["pitestRuntimeOnly"])
             from(project.files("${project.buildDir}/intermediates/sourceFolderJavaResources/${variant.dirName}"))

